### ISSUE TITLE
Can only run once after restarting my phone

### ISSUE THREAD

**Issue by SainoNamkho** [MuntashirAkon/AppManager#1051]
### Please check before submitting an issue

- [X] I know what my device, OS and App Manager versions are
- [X] I know how to take logs
- [X] I know how to reproduce the issue which may not be specific to my device

### Describe the bug

The AppManager app works in the first run after restarting my phone, in wireless adb mode.

Once after I killed the app (by swiping the app's card off the Recents view, which closes an app in MIUI), it takes a desperately long time (up to half an hour) stucking at the splahscreen to open, saying "Initializing"

![Screenshot_2023-04-24-14-10-01-479_io github muntashirakon AppManager debug](https://user-images.githubusercontent.com/23036788/233915969-4b14331b-6e7c-4275-81c0-1d536b02b631.jpg)

After that long wait, it asks me to chose what to do with wireless debugging.
![Screenshot_2023-04-24-14-32-27-386_io github muntashirakon AppManager debug-edit](https://user-images.githubusercontent.com/23036788/233925951-f4166c65-aba8-4ca7-9193-54d8ba27f8b3.jpg)

I chose to `connect`, only to wait another such long time before entering the app.

The `am.log` shows
```
2023-04-24 14:09:59.517 D/SplashActivity: Waiting to be authenticated.
2023-04-24 14:09:59.520 W/KSManager: Using keystore BKS
2023-04-24 14:09:59.526 I/CompatUtil: Loading local protection key
2023-04-24 14:09:59.529 I/CompatUtil: AES local protection key found in keystore
2023-04-24 14:09:59.561 D/SplashActivity: Authenticated
2023-04-24 14:09:59.562 D/SecurityAndOpsViewModel: Before Ops::init
2023-04-24 14:09:59.570 D/IPC: Init: Local server
2023-04-24 14:09:59.609 D/ServiceConnectionWrapper: Launching service...
2023-04-24 14:10:44.611 E/CHECK
android.os.RemoteException: Binder not running.
	at io.github.muntashirakon.AppManager.ipc.ServiceConnectionWrapper.getService(ServiceConnectionWrapper.java:86)
	at io.github.muntashirakon.AppManager.ipc.ServiceConnectionWrapper.bindService(ServiceConnectionWrapper.java:98)
	at io.github.muntashirakon.AppManager.ipc.LocalServices.bindAmService(LocalServices.java:56)
	at io.github.muntashirakon.AppManager.servermanager.LocalServer.launchAmService(LocalServer.java:66)
	at io.github.muntashirakon.AppManager.servermanager.LocalServer.getInstance(LocalServer.java:54)
	at io.github.muntashirakon.AppManager.settings.Ops.isAMServiceUpAndRunning(Ops.java:510)
	at io.github.muntashirakon.AppManager.settings.Ops.init(Ops.java:204)
	at io.github.muntashirakon.AppManager.settings.SecurityAndOpsViewModel.lambda$setModeOfOps$0$io-github-muntashirakon-AppManager-settings-SecurityAndOpsViewModel(SecurityAndOpsViewModel.java:75)
	at io.github.muntashirakon.AppManager.settings.SecurityAndOpsViewModel$$ExternalSyntheticLambda3.run(Unknown Source:2)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:463)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)
2023-04-24 14:10:44.622 D/SecurityAndOpsViewModel: After Ops::init
2023-04-24 14:10:59.714 W/LocalServerManager: closeBgServer: null  Read timed out
2023-04-24 14:15:12.367 D/SplashActivity: Try auto-connecting to wireless debugging.
2023-04-24 14:15:12.370 D/SecurityAndOpsViewModel: Before Ops::autoConnectAdb
2023-04-24 14:15:47.434 W/LocalServerManager: closeBgServer: null  Read timed out
2023-04-24 14:16:32.454 E/ADB
android.os.RemoteException: Binder not running.
	at io.github.muntashirakon.AppManager.ipc.ServiceConnectionWrapper.getService(ServiceConnectionWrapper.java:86)
	at io.github.muntashirakon.AppManager.ipc.ServiceConnectionWrapper.bindService(ServiceConnectionWrapper.java:98)
	at io.github.muntashirakon.AppManager.ipc.LocalServices.bindAmService(LocalServices.java:56)
	at io.github.muntashirakon.AppManager.servermanager.LocalServer.launchAmService(LocalServer.java:66)
	at io.github.muntashirakon.AppManager.servermanager.LocalServer.restart(LocalServer.java:196)
	at io.github.muntashirakon.AppManager.settings.Ops.autoConnectAdb(Ops.java:355)
	at io.github.muntashirakon.AppManager.settings.SecurityAndOpsViewModel.lambda$autoConnectAdb$1$io-github-muntashirakon-AppManager-settings-SecurityAndOpsViewModel(SecurityAndOpsViewModel.java:86)
	at io.github.muntashirakon.AppManager.settings.SecurityAndOpsViewModel$$ExternalSyntheticLambda0.run(Unknown Source:4)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:463)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)
2023-04-24 14:16:32.461 D/SecurityAndOpsViewModel: After Ops::autoConnectAdb
```

Then I found it has no permission to display app ops and the `Running Apps` shows only AppManager itself. I checked the mode of operation, which is `Wireless debugging (inferred mode: no root)`.

Again, after I re-applied the wireless debugging mode, it comes to an infinite loading.

https://user-images.githubusercontent.com/23036788/233927581-dc68cd86-81f7-4424-a7a7-cbf2b5b3ee31.mp4

Sometimes the loading takes forever, sometimes it ends after minutes with mode set to `Wireless debugging`.
But it does not show any app info, saying "App not installed" and "Counld not fetch package info". If I come back to the settings page, the mode shows `Wireless debugging (inferred mode: no root)` or `Wireless debugging (inferred mode: ADB)`

https://user-images.githubusercontent.com/23036788/233928400-20144381-46e1-4087-bd9b-093933a287b5.mp4

### To Reproduce

I've mentioned the steps in the _Descibe the bug_ part.

### Logs

The final `am.log` is 
[am.log](https://github.com/MuntashirAkon/AppManager/files/11307612/am.log)

Here's another copy, in case it's more convenient to view: https://pastebin.ubuntu.com/p/kbypmFZFBJ/

### Device info

- Device: Xiaomi Redmi K50 Ultra (known as Xiaomi 12T Pro overseas mainland China)
- OS Version: MIUI 14.0.23.3.20, Android 13
- App Manager Version: 3.1.0 stable and 3.1.0-2315-DEBUG
- Mode: ADB

### Additional context

1. Only restarting the phones counts, clearing app data or resintalling it makes no difference.
2. I guess MIUI does not remember AppManager correctly after pairing the wireless debugging. Some indications:
    1. This problem especially happens when I reopened WLAN.
    2. Unlike other apps, I cannot forget AppManger from paired devices without forgetting another device together.

    https://user-images.githubusercontent.com/23036788/233923285-99480a5f-b463-4c5e-b71c-e893956ccde6.mp4

**Comment by SainoNamkho** [comment]
Sorry that the message is truncated becauses it's too long, here's the truncated text (after the end of _Describe the bug_ part).

After that long wait, it asks me to chose what to do with wireless debugging.
![Screenshot_2023-04-24-14-32-27-386_io github muntashirakon AppManager debug-edit](https://user-images.githubusercontent.com/23036788/233925951-f4166c65-aba8-4ca7-9193-54d8ba27f8b3.jpg)

I chose to `connect`, only to wait another such long time before entering the app.

The `am.log` shows
```
2023-04-24 14:09:59.517 D/SplashActivity: Waiting to be authenticated.
2023-04-24 14:09:59.520 W/KSManager: Using keystore BKS
2023-04-24 14:09:59.526 I/CompatUtil: Loading local protection key
2023-04-24 14:09:59.529 I/CompatUtil: AES local protection key found in keystore
2023-04-24 14:09:59.561 D/SplashActivity: Authenticated
2023-04-24 14:09:59.562 D/SecurityAndOpsViewModel: Before Ops::init
2023-04-24 14:09:59.570 D/IPC: Init: Local server
2023-04-24 14:09:59.609 D/ServiceConnectionWrapper: Launching service...
2023-04-24 14:10:44.611 E/CHECK
android.os.RemoteException: Binder not running.
	at io.github.muntashirakon.AppManager.ipc.ServiceConnectionWrapper.getService(ServiceConnectionWrapper.java:86)
	at io.github.muntashirakon.AppManager.ipc.ServiceConnectionWrapper.bindService(ServiceConnectionWrapper.java:98)
	at io.github.muntashirakon.AppManager.ipc.LocalServices.bindAmService(LocalServices.java:56)
	at io.github.muntashirakon.AppManager.servermanager.LocalServer.launchAmService(LocalServer.java:66)
	at io.github.muntashirakon.AppManager.servermanager.LocalServer.getInstance(LocalServer.java:54)
	at io.github.muntashirakon.AppManager.settings.Ops.isAMServiceUpAndRunning(Ops.java:510)
	at io.github.muntashirakon.AppManager.settings.Ops.init(Ops.java:204)
	at io.github.muntashirakon.AppManager.settings.SecurityAndOpsViewModel.lambda$setModeOfOps$0$io-github-muntashirakon-AppManager-settings-SecurityAndOpsViewModel(SecurityAndOpsViewModel.java:75)
	at io.github.muntashirakon.AppManager.settings.SecurityAndOpsViewModel$$ExternalSyntheticLambda3.run(Unknown Source:2)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:463)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)
2023-04-24 14:10:44.622 D/SecurityAndOpsViewModel: After Ops::init
2023-04-24 14:10:59.714 W/LocalServerManager: closeBgServer: null  Read timed out
2023-04-24 14:15:12.367 D/SplashActivity: Try auto-connecting to wireless debugging.
2023-04-24 14:15:12.370 D/SecurityAndOpsViewModel: Before Ops::autoConnectAdb
2023-04-24 14:15:47.434 W/LocalServerManager: closeBgServer: null  Read timed out
2023-04-24 14:16:32.454 E/ADB
android.os.RemoteException: Binder not running.
	at io.github.muntashirakon.AppManager.ipc.ServiceConnectionWrapper.getService(ServiceConnectionWrapper.java:86)
	at io.github.muntashirakon.AppManager.ipc.ServiceConnectionWrapper.bindService(ServiceConnectionWrapper.java:98)
	at io.github.muntashirakon.AppManager.ipc.LocalServices.bindAmService(LocalServices.java:56)
	at io.github.muntashirakon.AppManager.servermanager.LocalServer.launchAmService(LocalServer.java:66)
	at io.github.muntashirakon.AppManager.servermanager.LocalServer.restart(LocalServer.java:196)
	at io.github.muntashirakon.AppManager.settings.Ops.autoConnectAdb(Ops.java:355)
	at io.github.muntashirakon.AppManager.settings.SecurityAndOpsViewModel.lambda$autoConnectAdb$1$io-github-muntashirakon-AppManager-settings-SecurityAndOpsViewModel(SecurityAndOpsViewModel.java:86)
	at io.github.muntashirakon.AppManager.settings.SecurityAndOpsViewModel$$ExternalSyntheticLambda0.run(Unknown Source:4)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:463)
	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
	at java.lang.Thread.run(Thread.java:1012)
2023-04-24 14:16:32.461 D/SecurityAndOpsViewModel: After Ops::autoConnectAdb
```

Then I found it had no permission to display app ops and the `Running Apps` shows only AppManager itself. I checked the mode of operation, which is `Wireless debugging (inferred mode: no root)`.

Again, after I re-applied the wireless debugging mode, it comes to an infinite loading.

https://user-images.githubusercontent.com/23036788/233927581-dc68cd86-81f7-4424-a7a7-cbf2b5b3ee31.mp4

Sometimes the loading takes forever, sometimes it ends after minutes with mode set to `Wireless debugging`.
But it does not show any app info, saying "App not installed" and "Counld not fetch package info". If I come back to the settings page, the mode shows `Wireless debugging (inferred mode: no root)` or `Wireless debugging (inferred mode: ADB)`

https://user-images.githubusercontent.com/23036788/233928400-20144381-46e1-4087-bd9b-093933a287b5.mp4

**Comment by SainoNamkho** [comment]
Maybe related: #948

**Comment by MuntashirAkon** [comment]
After the first run, `am_local_server` daemon should be running in UID `2000` (killing App Manager does not kill this daemon, only killing the ADB daemon i.e. `adbd` would kill this process). As a result, running App Manager again should directly connect to the daemon instead of attempting to connect to ADB. Logs are saying that this didn't happen (no broadcast was delivered to App Manager indicating whether the current service is alive). So, I guess  this could be the primary issue. You can check whether this process exists via ADB Shell using the following command:

```sh
ps -A | grep am_local_server
```

If this process exist, you need to send me the full system logs when you launch App Manager. This should log why it failed to deliver the broadcast to App Manager.

**Comment by MuntashirAkon** [comment]
```
2023-04-24 14:16:32.461 D/SecurityAndOpsViewModel: After Ops::autoConnectAdb
2023-04-24 14:32:17.891 D/SplashActivity: Display wireless debugging chooser (pair or connect)
```

There's almost 16 minutes gap between these two messages which should only occur if the app was paused during that period.

**Comment by SainoNamkho** [comment]
Seems fixed in version 4.0.0

**Comment by MuntashirAkon** [comment]
Thanks.

**Comment by ascendbeing** [comment]
I'm having this issue @MuntashirAkon . I just got  Samsung OTA maybe it's related? currently not able to use adb modes. I was able to pair but not ever start it with adb mode. I'm on the latest action for master branch. I've tried toggling adb, stopping am, stopping shizuku. literally won't let me use adb mode it just hangs and if I'm lucky it's normal mode. it might fix itself with a restart idk, haven't tested this Ota Maybe it killed it for goof
